### PR TITLE
Simplify vouch and lighthouse configurations

### PR DESCRIPTION
**Pull request by dB2510**
Adds a section for vouch integration in README. It simplifies vouch.yml, vouch/run.sh configurations. Remove usage of dockerfile for lighthouse.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,6 +141,25 @@ Keep checking in for updates, [here](https://github.com/ObolNetwork/charon/#supp
...
+- The `vouch/` directory contains a basic vouch config file `vouch.yml` which is programmatically populated by `run.sh`.
+- `Dockerfile` is used for building vouch image which uses `run.sh` as entrypoint.
+- The `run.sh` script:
+    - Creates an [ethdo](https://github.com/wealdtech/ethdo) wallet which imports the keystores created during dkg ceremony or with `charon create cluster` command.
+    - Adds the `beacon-node-address` for vouch VC to use. This is obtained from `VOUCH_BEACON_NODE_ADDRESS` environment variable in `docker-compose.yml`.
+    - Starts the validator client.
+
+### Why is custom config required?
+- Vouch requires a [configuration file](https://github.com/attestantio/vouch/blob/master/docs/configuration.md#the-configuration-file) called `vouch.yml` (or `vouch.json`). This config requires certain fields that are specific to running a instance of vouch.
+- For ex, it requires a top-level `beacon-node-address` field which is different for different charon node. Ex: http://node0:3600 for node0, http://node1:3600 for node1 etc.
```
Comment (xenowits) [inline README.md:158]:
```suggestion
- For example, it requires a top-level `beacon-node-address` field which is different for different charon node. Ex: http://node0:3600 for node0, http://node1:3600 for node1 etc.
```

File: README.md [hunk 2]
```diff
@@ -141,6 +141,25 @@ Keep checking in for updates, [here](https://github.com/ObolNetwork/charon/#supp
...
+- Timeout is set to 10s in `vouch.yml` for all the strategies supported to allow fetching duties for DVT.
```
Comment (xenowits) [inline README.md:159]:
```suggestion
- Timeout is set to 10s in `vouch.yml` for all the strategies supported to allow sufficient time for fetching duties for DVT.
```

File: vouch/run.sh [hunk 3]
```diff
@@ -21,18 +21,10 @@ wallet="validators"
...
-  wallet=${wallet} yq -i '.accountmanager.wallet.accounts[0] = strenv(wallet)' ${vouchFile}
+  cp /opt/charon/vouch/vouch.yml ${vouchFile}
 
   # Set beacon node addresses.
-  nodeAddr="http://${NODE}:3600"
-  addr=${nodeAddr} yq -i '.beacon-node-address = strenv(addr)' ${vouchFile}
-  addr=${nodeAddr} yq -i '.submitter.beacon-node-addresses[0] = strenv(addr)' ${vouchFile}
-  addr=${nodeAddr} yq -i '.strategies.beaconblockproposal.beacon-node-addresses[0] = strenv(addr)' ${vouchFile}
-  addr=${nodeAddr} yq -i '.strategies.attestationdata.beacon-node-addresses[0] = strenv(addr)' ${vouchFile}
+  addr=${VOUCH_BEACON_NODE_ADDRESS} yq -i '.beacon-node-address = strenv(addr)' ${vouchFile}
```
Comment (corverroos) [inline vouch/run.sh:27]: vouch can read this env var directly, no need to set in config file

File: vouch/run.sh [hunk 4]
```diff
@@ -5,54 +5,34 @@
...
 # Import keys into the ethdo wallet.
 account=0
-for f in /opt/charon/node/validator_keys/keystore-*.json; do
+for f in /opt/validator_keys/keystore-*.json; do
   accountName="account-${account}"
   echo "Importing key ${f} into ethdo wallet: ${wallet}/${accountName}"
 
-  PASSPHRASE=$(cat "${f//json/txt}")
+  KEYSTORE_PASSPHRASE=$(cat "${f//json/txt}")
+  ACCOUNT_PASSPHRASE="T8BFYJZU5R" # Hardcoded ethdo account passphrase
```
Comment (xenowits) [inline vouch/run.sh:29]: move this to top

File: vouch/vouch.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+# Address of the charon node. Ex: http://node0:3600
+beacon-node-address:
+
+# The wallet account manager obtains account information from local wallets, and signs locally.
+# It supports wallets created by ethdo.
+accountmanager:
+  wallet:
+    locations: /opt/vouch/keys
+    accounts: validators
+    passphrases: T8BFYJZU5R
```
Comment (xenowits) [inline vouch/vouch.yml:14]: add comment here: `Passphrase should match ACCOUNT_PASSPHRASE in run.sh`

File: vouch/vouch.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+# metrics is the module that logs metrics, in this case using prometheus. Note that vouch doesn't emit metrics if
+# the following block is not provided.
+metrics:
```
Comment (xenowits) [inline vouch/vouch.yml:18]: add this to `prometheus.yml` if not present there

File: vouch/run.sh [hunk 7]
```diff
@@ -62,10 +42,10 @@ done
 # Log wallet info.
 echo "Starting vouch validator client. Wallet info:"
 /app/ethdo \
---base-dir=/opt/charon/keys/ethdo wallet info \
+--base-dir=${keysDir} wallet info \
 --wallet="${wallet}" \
---base-dir="${baseDir}" \
+--base-dir="${keysDir}" \
```
Comment (xenowits) [inline vouch/run.sh:47]: check if both `--both-dir` flags are required

**Review by xenowits — APPROVED** [review]
Looks lean 🚀

File: vouch/vouch.yml (hunk 6 above)
Comment (dB2510) [reply vouch/vouch.yml:18]: done

File: lighthouse/run.sh [hunk 8]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env bash
 
-while ! curl "http://${NODE}:3600/eth/v1/node/health" 2>/dev/null; do
-  echo "Waiting for http://${NODE}:3600 to become available..."
+apt-get update && apt-get install -y curl jq wget
```
Comment (corverroos) [inline lighthouse/run.sh:3]: so this is introduces a delay of 1m to every time you start the  lighthouse container. Not sure this optimisation is worth it.

File: vouch/run.sh [hunk 9]
```diff
@@ -5,54 +5,34 @@
...
-
-# Remove directory if it already exists.
-rm -r ${baseDir} || true
-
-# Create a fresh directory for ethdo keys.
-mkdir ${baseDir}
+BASE_DIR="/opt/vouch"
+VOUCH_FILE="/opt/vouch/vouch.yml"
+KEYS_DIR="/opt/vouch/keys"
+ACCOUNT_PASSPHRASE="T8BFYJZU5R" # Hardcoded ethdo account passphrase
```
Comment (corverroos) [inline vouch/run.sh:11]: suggest "secret" to make it obvious this is not a proper secret.

File: vouch/vouch.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
+# This is a sample config to run vouch VC. It is used by the run.sh script
+# to generate a custom config for each vouch VC connected to a charon node.
+# Refer: https://github.com/attestantio/vouch/blob/master/docs/configuration.md.
+
+# Address of the charon node. Ex: http://node0:3600
+beacon-node-address:
```
Comment (corverroos) [inline vouch/vouch.yml:6]: remove this

File: vouch/vouch.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    timeout: 10s
+  attestationdata:
+    timeout: 10s
+  aggregateattestation:
+    timeout: 10s
+  synccommitteecontribution:
+    timeout: 10s
+
+feerecipient:
+  default-address: '0x0000000000000000000000000000000000000001'
```
Comment (corverroos) [inline vouch/vouch.yml:37]: can we remove these addresses? or at least add a comment why dummy these values are correct

File: vouch/Dockerfile [hunk 12]
```diff
@@ -11,4 +11,4 @@ ENV YQ_BINARY=yq_linux_amd64
 RUN wget https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/${YQ_BINARY} -O /usr/bin/yq \
```
Comment (corverroos) [inline vouch/Dockerfile:11]: I think we can get by without yq

File: vouch/vouch.yml (hunk 11 above)
Comment (dB2510) [reply vouch/vouch.yml]: `fallback-fee-recipient` is required in blockrelay, added comment

**Review by corverroos — APPROVED** [review]

**Pull request merged**